File: app/src/main/java/com/view/jameson/androidrecyclerviewcard/CardAdapter.java
package com.view.jameson.androidrecyclerviewcard;

import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;

import com.view.jameson.library.CardAdapterHelper;

import java.util.ArrayList;
import java.util.List;

import jameson.io.library.util.ToastUtils;

/**
 * Created by jameson on 8/30/16.
 */
class CardAdapter extends RecyclerView.Adapter<CardAdapter.ViewHolder> {
    private List<Integer> mList;


    private CardAdapterHelper mCardAdapterHelper = new CardAdapterHelper();

    CardAdapter(List<Integer> mList, int mPagePadding, int mShowLeftCardWidth) {
        this.mList = mList;
        mCardAdapterHelper.setPagePadding(mPagePadding);
        mCardAdapterHelper.setShowLeftCardWidth(mShowLeftCardWidth);
    }

    @Override
    public ViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
        View itemView = LayoutInflater.from(parent.getContext()).inflate(R.layout.view_card_item, parent, false);
        mCardAdapterHelper.onCreateViewHolder(parent, itemView);
        return new ViewHolder(itemView);
    }

    @Override
    public void onBindViewHolder(final ViewHolder holder, final int position) {
        mCardAdapterHelper.onBindViewHolder(holder.itemView, position, getItemCount());
        holder.mImageView.setImageResource(mList.get(position));
        holder.mImageView.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                ToastUtils.show(holder.mImageView.getContext(), "" + position);
            }
        });
    }

    @Override
    public int getItemCount() {
        return mList.size();
    }

    public class ViewHolder extends RecyclerView.ViewHolder {
        public final ImageView mImageView;

        public ViewHolder(final View itemView) {
            super(itemView);
            mImageView = itemView.findViewById(R.id.imageView);
        }

    }

}

File: app/src/main/java/com/view/jameson/androidrecyclerviewcard/CardAdapter.kt
package com.view.jameson.androidrecyclerviewcard

import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ImageView
import androidx.recyclerview.widget.RecyclerView

import com.view.jameson.library.CardAdapterHelper



/**
 * Created by jameson on 8/30/16.
 */
class CardAdapter(private val mList: List<Int>, mPagePadding: Int, mShowLeftCardWidth: Int) : RecyclerView.Adapter<CardAdapter.ViewHolder>() {


    private val mCardAdapterHelper = CardAdapterHelper()

    init {
        mCardAdapterHelper.setPagePadding(mPagePadding)
        mCardAdapterHelper.setShowLeftCardWidth(mShowLeftCardWidth)
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
        val itemView = LayoutInflater.from(parent.context).inflate(R.layout.view_card_item, parent, false)
        mCardAdapterHelper.onCreateViewHolder(parent, itemView)
        return ViewHolder(itemView)
    }

    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
        mCardAdapterHelper.onBindViewHolder(holder.itemView, position, itemCount)
        holder.mImageView.setImageResource(mList[position])
        holder.mImageView.setOnClickListener {
//            ToastUtils.show(holder.mImageView.context, "" + position)
        }
    }

    override fun getItemCount(): Int {
        return mList.size
    }

    inner class ViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
        val mImageView: ImageView

        init {
            mImageView = itemView.findViewById(R.id.imageView)
        }

    }

}
